src/Microsoft.Data/PrimitiveColumnContainer.cs

```diff
@@ -14,7 +16,7 @@ namespace Microsoft.Data
     /// PrimitiveDataFrameColumnContainer is just a store for the column data. APIs that want to change the data must be defined in PrimitiveDataFrameColumn
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    internal partial class PrimitiveColumnContainer<T> : IEnumerable<T?>
+    internal partial class PrimitiveColumnContainer<T> : IEnumerable<ReadOnlyMemory<T>>
```

Comment:
I'm not sure if this should be a `IEnumerable<ReadOnlyMemory<T>>` or a `IEnumerable<Memory<T>>`. A `PrimitiveColumnContainer` could hold either `ReadOnlyMemory<T>` or `Memory<T>` in its `Buffers`. `IEnumerable<ReadOnlyMemory<T>>` would work for either, but it'd be immutable. If we instead always expose an `IEnumerable<Memory<T>>`, it'd involve an unnecessary copy for simple enumeration use cases. 

Another option is to not touch the interface but add 2 new APIs that return a `IEnumerable<ReadOnlyMemory<T>>` and an `IEnumerable<Memory<T>>`. That way, the users of the API have options.

Reply:
Honestly, I think a new API (or two) is called for here. Something like `GetBuffers()` and `GetReadOnlyBuffers()`?

---
In reply to: [321517883](https://github.com/dotnet/corefxlab/pull/2718#discussion_r321517883) [](ancestors = 321517883)